build/gulpfile.vscode.js

```diff
@@ -495,15 +502,18 @@ BUILD_TARGETS.forEach(buildTarget => {
 			tasks.push(patchWin32DependenciesTask(destinationFolderName));
 		}
 
-		const vscodeTaskCI = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}-ci`, task.series(...tasks));
+		const vscodeTaskCI = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}-ci`, task.series(
+			compileNativeExtensionsBuildTask,
+			...tasks
+		));
 		gulp.task(vscodeTaskCI);
 
 		const vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
 			compileBuildTask,
 			compileExtensionsBuildTask,
 			compileExtensionMediaBuildTask,
 			minified ? minifyVSCodeTask : bundleVSCodeTask,
-			vscodeTaskCI
+			...tasks
```

Comment:
Why is this changed ? Don't we want `compileNativeExtensionsBuildTask` in this case ?

Reply:
`compileExtensionsBuildTask` builds everything so I didn't want to duplicate work

Reply:
I see, if we change the logic per https://github.com/microsoft/vscode/pull/233739/files#r1839347402 then we can avoid this change as well. Basically, making `compileExtensionsBuildTask` and `compileNativeExtensionsBuildTask` mutually exclusive

Reply:
I like it, thanks!